cli/tests/integrations/test_experimental.py

```diff
@@ -0,0 +1,535 @@
+            'The package [{}] version [{}] has been added to DC/OS\n'.format(
+                name, version)).encode()
+        assert out == stdout, (out, stdout)
+
+    assert name == package_name
+    assert version == package_version if package_version else True
+
+    return name, version
+
+
+def _package_build(build_definition_path,
+                   output_directory,
+                   metadata=None,
+                   manifest=None):
+    command = _package_build_cmd(build_definition_path,
+                                 output_directory)
+
+    code, out, err = exec_command(command)
+    assert code == 0
+    assert err == b'Created DCOS Universe package: '
+
+    package_path = out.decode().rstrip()
+    assert os.path.exists(package_path)
+
+    name, version, md5 = _decompose_name(package_path)
+    build_definition = _get_json(build_definition_path)
+    assert name == build_definition['name']
+    assert version == build_definition['version']
+    assert md5 == _get_md5_hash(package_path)
+    assert (manifest is None or
+            manifest == _get_json_in_zip(package_path, 'manifest.json'))
+    assert (metadata is None or
+            metadata == _get_json_in_zip(package_path, 'metadata.json'))
+
+    return package_path
+
+
+def _package_build_failure(build_definition_path,
+                           output_directory,
+                           return_code=1,
+                           stdout=b'',
+                           stderr=b''):
+    command = _package_build_cmd(build_definition_path, output_directory)
+
+    assert_command(command,
+                   returncode=return_code,
+                   stdout=stdout,
+                   stderr=stderr)
+    assert len(os.listdir(output_directory)) == 0
+
+
+def _wait_for_package_add_remote(package_name, package_version):
+    command = _package_add_cmd(package_name=package_name,
+                               package_version=package_version)
+    _wait_for_package_add(command, package_name, package_version)
+
+
+def _wait_for_package_add_local(package):
+    command = _package_add_cmd(dcos_package=package)
+    metadata = _get_json_in_zip(package, 'metadata.json')
+    name = metadata['name']
+    version = metadata['version']
+    _wait_for_package_add(command, name, version)
+
+
+def _wait_for_package_add(command, name, version):
+    done_adding = False
+    max_retries = 10
+    retry_number = 0
+    while not done_adding:
+        code, out, err = exec_command(command)
+        change_in_progress_message = (
+            'A change to package '
+            '{}-{} is already in progress\n'.format(
+                name, version).encode())
+        done_adding = code == 0
+        assert (done_adding or
+                (code == 1 and
+                 err == change_in_progress_message))
+        assert retry_number != max_retries, \
+            'Waiting for package add to complete took too long'
+        retry_number += 1
+        time.sleep(5)
```

Comment:
Running on my machine, this seemed to be about double the amount it usually took to complete.